src/Ocelot/Security/IPSecurity/IPSecurityPolicy.cs

```diff
@@ -30,13 +30,12 @@ public Response Security(DownstreamRoute downstreamRoute, HttpContext context)
             if (!options.IPAllowedList.Contains(clientIp.ToString()))
             {
                 var error = new UnauthenticatedError($"{clientIp} does not allow access, the request is invalid");
-                return new ErrorResponse(error);
+                return ValueTask.FromResult<Response>(new ErrorResponse(error));
             }
         }
 
-        return new OkResponse();
+        return ValueTask.FromResult(OkResponse);
     }
 
-    public Task<Response> SecurityAsync(DownstreamRoute downstreamRoute, HttpContext context)
-        => Task.Run(() => Security(downstreamRoute, context));
```

Comment:
That is not good. I know. But don't remove!
And call async version from sync one aka `SecurityAsync().GetAwaiter().GetResult()`